instrumentation/redis/lib/opentelemetry/instrumentation/redis/utils.rb

```diff
@@ -17,25 +17,44 @@ module Utils
 
         def format_span_name(commands)
           commands.map do |command|
-            format_arg(resolve_command_args(command).first)
+            format_cmd(resolve_command_args(command).first)
           end.join(' ')
         end
 
         def format_statement(command_args)
           command_args = resolve_command_args(command_args)
           return 'AUTH ?' if auth_command?(command_args)
 
-          cmd = command_args.map { |x| format_arg(x) }.join(' ')
-          OpenTelemetry::Common::Utilities.truncate(cmd, CMD_MAX_LEN)
+          cmd = format_cmd(command_args.first)
+          args = command_args[1..-1].map { |arg| format_arg(arg) }
+          statement = args.unshift(cmd).join(' ')
+          OpenTelemetry::Common::Utilities.truncate(statement, CMD_MAX_LEN)
         end
 
         def format_statements(commands)
           commands.map { |command| format_statement(command) }.join("\n")
         end
 
+        def format_cmd(cmd)
+          format_string(cmd.to_s.upcase)
+        end
+
         def format_arg(arg)
-          str = arg.is_a?(Symbol) ? arg.to_s.upcase : arg.to_s
-          str = OpenTelemetry::Common::Utilities.utf8_encode(str, binary: true)
+          format_string(obfuscate_arg(arg.to_s))
+        end
+
+        private
+
+        def obfuscate_arg(arg)
+          if config[:enable_statement_obfuscation]
+            PLACEHOLDER
+          else
+            arg
+          end
+        end
+
+        def format_string(input = nil)
+          str = OpenTelemetry::Common::Utilities.utf8_encode(input, binary: true)
           OpenTelemetry::Common::Utilities.truncate(str, VALUE_MAX_LEN)
         rescue StandardError => e
```

Comment:
This error handling should be moved up to `format_arg` and `format_cmd`. This will resolve the test failure.

Reply:
sure

Reply:
Though we're interpolating the input in the debug message so I suppose we'd still run into this problem for `BasicObject`, but let's say that's an edge case.